### PR TITLE
Fixes contributors guide pull request doc #3087

### LINKED ISSUE #3087: The Contributor Guide pull request doc conflates developer and non-developer personas
/area contributor-guide
/kind documentation

Summarizing a [thread in #sig-contribex](https://kubernetes.slack.com/archives/C1TU9EB9S/p1547230614247500)

We view the contributor guide as a place where all contributors - developers, writers, etc can get the same information and if anything speaks to solely the developers persona, it should go in a dev guide.

We need to edit the contributor guide PR doc:
- A lot of folks who only want to contribute to our contributing docs and end up getting stuck on the dev environment pieces that they don't need.
- Need to add disclosures / separate code/non-code concerns
- There are also significant "setting up your dev environment" steps in there that should just live elsewhere, not really in a "how to pull request" doc!

(ref: https://github.com/kubernetes/community/issues/1413 for the origin story of the contrib guide)

https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md is the doc in question, some content probably belongs in:
- https://github.com/kubernetes/community/blob/master/contributors/devel/development.md
- https://github.com/kubernetes/community/blob/master/contributors/devel/testing.md
- [etiquette](https://github.com/kubernetes/community/issues/3032)

/help
/assign @guineveresaenger @parispittman 
Please /unassign and delegate as needed

### PR DESCRIPTION
**Pull request by Sai-Adarsh**
* Fixes #3087 
* Build contents from [/contributors/guide/github-worflow.md](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#build)  belongs to [/community/contributors/devel/development.md](https://github.com/kubernetes/community/blob/master/contributors/devel/development.md)

/assign @guineveresaenger @spiffxp

### CODE REVIEW

**Comment by guineveresaenger** [comment]
This is so much better, yay! 

Could you drop a redirect in https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#build
that says something like "This workflow is process-specific; for detailed build instructions for kubernetes/kubernetes please see here" and link?

**Comment by Sai-Adarsh** [comment]
> This is so much better, yay!
> 
> Could you drop a redirect in https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#build
> that says something like "This workflow is process-specific; for detailed build instructions for kubernetes/kubernetes please see here" and link?

@guineveresaenger done!

File: contributors/devel/development.md [hunk 1]
```diff
@@ -122,9 +122,96 @@ platform.
 
 ### etcd
 
-Kubernetes maintains state in [`etcd`][etcd-latest], a distributed key store.
```
Comment (guineveresaenger) [inline contributors/devel/development.md]: Why was this removed?

File: contributors/devel/development.md [hunk 2]
```diff
@@ -122,9 +122,96 @@ platform.
 
 ### etcd
```
Comment (guineveresaenger) [inline contributors/devel/development.md]: This is a header with no context

File: contributors/devel/development.md [hunk 3]
```diff
@@ -122,9 +122,96 @@ platform.
...
+# https://git.k8s.io/community/contributors/devel/sig-testing/e2e-tests.md#building-kubernetes-and-running-the-tests
+make test-e2e
+```
+
+See the [testing guide](/contributors/devel/sig-testing/testing.md) and [end-to-end tests](/contributors/devel/sig-testing/e2e-tests.md)
+for additional information and scenarios.
+
+Run `make help` for additional information on these make targets.
 
-Please [install it locally][etcd-install] to run local integration tests.
```
Comment (guineveresaenger) [inline contributors/devel/development.md]: this should probably stay

File: contributors/devel/development.md [hunk 4]
```diff
@@ -122,9 +122,96 @@ platform.
...
+# Equivalent to: go run hack/e2e.go -- -v --build --up --test --down
+# Note: running all e2e tests takes a LONG time! To run specific e2e tests, visit:
+# https://git.k8s.io/community/contributors/devel/sig-testing/e2e-tests.md#building-kubernetes-and-running-the-tests
+make test-e2e
+```
+
+See the [testing guide](/contributors/devel/sig-testing/testing.md) and [end-to-end tests](/contributors/devel/sig-testing/e2e-tests.md)
+for additional information and scenarios.
+
+Run `make help` for additional information on these make targets.
```
Comment (guineveresaenger) [inline contributors/devel/development.md]: This entire section should probably go below the Golang instructions.
Maybe have a `Prerequisites` header that has the etcd and Go stuff in it?

File: contributors/devel/development.md (hunk 2 above)
Comment (guineveresaenger) [reply contributors/devel/development.md]: Remove this header because it's part of the instructions you're moving.

File: contributors/devel/development.md (hunk 3 above)
Comment (guineveresaenger) [reply contributors/devel/development.md]: nevermind, it's part of the block of instructions, it should be deleted.

File: contributors/guide/github-workflow.md [hunk 5]
```diff
@@ -74,6 +74,8 @@ git checkout -b myfeature
 Then edit code on the `myfeature` branch.
 
 #### Build
+This workflow is process-specific; for detailed build instructions for [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) please [see here](https://github.com/kubernetes/community/blob/master/contributors/devel/development.md#building-kubernetes-on-a-local-osshell-environment).
```
Comment (guineveresaenger) [inline contributors/guide/github-workflow.md]: Use "quick start" instead of "Detailed" since that is literally how the next section starts

**Comment by guineveresaenger** [comment]
Hi @Sai-Adarsh - I apologize for the delay! I'm waiting on https://github.com/kubernetes/community/pull/3245 to get merged, which is working on this very doc, so you can ignore my suggestion for a Table of Content. Could you address just the `etcd` header confusion, and remove the instructions block from `github-workflow.md`? 

/hold

File: contributors/devel/development.md [hunk 6]
```diff
@@ -120,11 +120,98 @@ tools].
...
+make test-integration
+
+# Run e2e tests by building test binaries, turn up a test cluster, run all tests, and tear the cluster down
+# Equivalent to: go run hack/e2e.go -- -v --build --up --test --down
+# Note: running all e2e tests takes a LONG time! To run specific e2e tests, visit:
+# https://git.k8s.io/community/contributors/devel/sig-testing/e2e-tests.md#building-kubernetes-and-running-the-tests
+make test-e2e
+```
+
+See the [testing guide](/contributors/devel/sig-testing/testing.md) and [end-to-end tests](/contributors/devel/sig-testing/e2e-tests.md)
```
Comment (guineveresaenger) [inline contributors/devel/development.md]: this would now be in the same folder, so the link should just be `e2e-tests.md`

File: contributors/guide/github-workflow.md [hunk 7]
```diff
@@ -74,6 +74,8 @@ git checkout -b myfeature
 Then edit code on the `myfeature` branch.
 
 #### Build
+This workflow is process-specific; for quick start build instructions for [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) please [see here](https://github.com/kubernetes/community/blob/master/contributors/devel/development.md#building-kubernetes-on-a-local-osshell-environment).
```
Comment (guineveresaenger) [inline contributors/guide/github-workflow.md]: The first link should use `https://git.k8s.io/kubernetes`
For the second link, you can just use a relative link `/contributors/devel/development.md#building-kubernetes-on-a-local-osshell-environment` (double check me on this I'm really bad at links 😛 )

**Comment by Sai-Adarsh** [comment]
@guineveresaenger done!!

File: contributors/devel/development.md [hunk 8]
```diff
@@ -160,6 +154,97 @@ images.
...
+make test-integration
+
+# Run e2e tests by building test binaries, turn up a test cluster, run all tests, and tear the cluster down
+# Equivalent to: go run hack/e2e.go -- -v --build --up --test --down
+# Note: running all e2e tests takes a LONG time! To run specific e2e tests, visit:
+# https://git.k8s.io/community/contributors/devel/sig-testing/e2e-tests.md#building-kubernetes-and-running-the-tests
+make test-e2e
+```
+
+See the [testing guide](/contributors/devel/sig-testing/e2e-tests.md) and [end-to-end tests](/contributors/devel/sig-testing/e2e-tests.md)
```
Comment (guineveresaenger) [inline contributors/devel/development.md]: I think you changed the wrong link here 😅 

the testing guide is `testing.md`, please change that back?

and both links should be just `./testing.md` and `./e2e-tests.md`.

File: contributors/devel/development.md [hunk 9]
```diff
@@ -160,6 +154,97 @@ images.
...
+make test WHAT=./pkg/api/helper GOFLAGS=-v
+
+# Run integration tests, requires etcd
+# For more info, visit https://git.k8s.io/community/contributors/devel/sig-testing/testing.md#integration-tests
+make test-integration
+
+# Run e2e tests by building test binaries, turn up a test cluster, run all tests, and tear the cluster down
+# Equivalent to: go run hack/e2e.go -- -v --build --up --test --down
+# Note: running all e2e tests takes a LONG time! To run specific e2e tests, visit:
+# https://git.k8s.io/community/contributors/devel/sig-testing/e2e-tests.md#building-kubernetes-and-running-the-tests
```
Comment (guineveresaenger) [inline contributors/devel/development.md]: Since we're updating links, this link can be relative too:
`./e2e-tests.md#building-kubernetes-and-running-the-tests`

File: contributors/guide/github-workflow.md [hunk 10]
```diff
@@ -74,6 +74,8 @@ git checkout -b myfeature
 Then edit code on the `myfeature` branch.
 
 #### Build
+This workflow is process-specific; for quick start build instructions for [kubernetes/kubernetes](https://git.k8s.io/kubernetes) please [see here](/contributors/devel/development.md#building-kubernetes-on-a-local-osshell-environment).
+
 The following section is a quick start on how to build Kubernetes locally, for more detailed information you can see [kubernetes/build](https://git.k8s.io/kubernetes/build/README.md).
```
Comment (guineveresaenger) [inline contributors/guide/github-workflow.md]: Please remove this section from this document.

**Comment by Sai-Adarsh** [comment]
> @guineveresaenger done!!

shouldn't it be `/sig-testing/testing.md` and ` /sig-testing/e2e-tests.md`

**Comment by guineveresaenger** [comment]
I'm sorry if I wasn't clear in what section should be deleted from `github-workflow.md`!

Here's what I mean: 

1. delete lines 77 - 166 from `github-workflow.md`. We don't want duplicate content.

2. Under the `Build` headline in `github-workflow.md`, write a short sentence explaining where to find quickstart build instructions (I liked what you had there).

**Comment by Sai-Adarsh** [comment]
> I'm sorry if I wasn't clear in what section should be deleted from `github-workflow.md`!
> 
> Here's what I mean:
> 
> 1. delete lines 77 - 166 from `github-workflow.md`. We don't want duplicate content.
> 2. Under the `Build` headline in `github-workflow.md`, write a short sentence explaining where to find quickstart build instructions (I liked what you had there).

pushed the changes! 👍

**Comment by guineveresaenger** [comment]
Awesome!

Could you [squash your commits](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git) and then we can 🚢 !
